pkg/js/transformer/jsontodsl.ts

```diff
@@ -121,6 +121,18 @@ function parseSubRelation(
     return parseTupleToUserset(relationDefinition);
   }
 
+  if (relationDefinition.union != null) {
+    return `(${parseUnion(typeName, relationName, relationDefinition, typeRestrictions)})`;
+  }
+
+  if (relationDefinition.intersection != null) {
+    return `(${parseIntersection(typeName, relationName, relationDefinition, typeRestrictions)})`;
+  }
+
+  if (relationDefinition.difference != null) {
+    return `(${parseDifference(typeName, relationName, relationDefinition, typeRestrictions)})`;
+  }
+
   throw new UnsupportedDSLNestingError(typeName, relationName);
```

Comment:
This error now requires an update.

Reply:
Yep, needs to handle the cases we aren't currently supporting (see JSON in #113 )